packages/portfolio-contract/src/portfolio.contract.ts

```diff
@@ -202,10 +202,17 @@ export const contract = async (
   if (!('axelar' in chainInfo)) {
     trace('⚠️ no axelar chainInfo; GMP not available', Object.keys(chainInfo));
   }
-  // TODO: only on 1st incarnation
-  registerChainsAndAssets(chainHub, brands, chainInfo, assetInfo, {
-    log: trace,
-  });
+
+  const baggage = zone.mapStore('baggage');
+  const firstIncarnationKey = 'firstIncarnationKey';
+
+  // Only register chains and assets on first incarnation to avoid conflicts on restart
+  if (!baggage.has(firstIncarnationKey)) {
```

Comment:
How about instead the condition is whether `chainHub` is empty? The current getters are async (due to https://github.com/Agoric/agoric-sdk/issues/10602 ) but we could add an `isEmpty` method that is synchronous.

Reply:
would looking for the chainhub itself work?

```suggestion
  if (!baggage.has('chainHub')) {
```

Reply:
I had Sonnet add that and I've pushed the change.

Reply:
I didn't see dckc's comment before pushing isEmpty. But to answer the question, `chainHub` is in the baggage on first incarnation so that check wouldn't work. (I think dckc pointed this out too in Slack; I'm just replying here for other readers)